eclipse-theia/theia
# [CI] Automate license check reviews

#### What it does

Introduce automated license check reviews
    
In this PR we introduce the option to use our license check tool,
"dash-licenses", in "Automatic IP Team Review Requests" mode [1].
In this mode, any dependency that's found to have an unclear or
suspicious license will be automatically submitted to the Eclipse
Foundation for review. Each such dependency will have a ticket opened
on the Foundation's Gitlab and be automatically reviewed. If the
automated review is not conclusive, a manual assessment will be
performed by the Foundation's IP team.
    
In our experience, most dependencies are approved automatically
within minutes.
    
To perform a license check with automated reviews, use the new script:
> $> yarn license:check:review
    
To perform the license check without the automated review, do as before:
> $> yarn license:check
    
Note: for review mode to work, a Personal Access Token from the Foundation's
Gitlab is required, created from an Eclipse committer's Gitlab profile. I went ahead and
set one as a GitHub secret, for this repo. 
Set it in an environment variable named "DASH_LICENSES_PAT". E.g. in bash:
>$> export DASH_LICENSES_PAT=\<token\>

[1] https://github.com/eclipse/dash-licenses#automatic-ip-team-review-requests

P.S. we obtained [permission](https://github.com/eclipse/dash-licenses/issues/129#issuecomment-1013329932) to do this a while ago.

#### How to test


- Locally, with and without a PAT set, confirm that `yarn license:check` still works like before
- Locally, with and without a PAT set, confirm that the new `yarn license:check:review ` works as expected:
    - without PAT: the tool exits immediately with an error message about the missing token
    - with PAT: the tool proceeds to run and creates or points-to existing Gitlab issues about the problematic dependencies. 

_`"btoa": "1.2.1"` can be used currently as a suspicious dependency for testing purposes - add it to root `package.json` and run `yarn`_

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
